sunpy/image/slit.py

```diff
@@ -0,0 +1,130 @@
+import sunpy
+import sunpy.map
+import sunpy.cm
+import sunpy.wcs
+
+import numpy as np
+import astropy.wcs
+import astropy.io.fits as fits
+import astropy.units as u
+
+import scipy.ndimage as snd
+import matplotlib.pyplot as plt
+from glob import glob
+
+
+
+def slit(mcube_in, x1, y1, x2, y2):
+    """
+    Returns an array with intensity along the slit on the y axis and time 
+    along x.
+    
+    Parameters    
+    ----------
+    in_files : `sunpy.map.MapCube`
+        A mapcube of the images you want to perform the slit analysis on. 
+        Usually with x and y as space, and z as time.
+    x1, y1 : `astropy.units.Quantity` 
+        The x and y coordinates of the beginning of the slit. May be either 
+        pixel coordinates or astropy units.
+    x2, y2 : `astropy.units.Quantity`
+        The x and y coordinates of the end of the slit. May be either 
+        pixel coordinates or astropy units.
+
+        
+    Returns
+    -------
+    out : `numpy.ndarray`
+        A numpy array with intenisty in y and time in x.
+
+    """
+    
+    # check the attributes of the coordinates
+    if ((isinstance(x1 and y1, u.Quantity) and isinstance(x2 and y2, u.Quantity)) or
+        (hasattr(x1 and y1, 'unit') and hasattr(x2 and y2, 'unit'))):
+    
+        if (x1.unit.is_equivalent(mcube_in[0].units.x) and
+            y1[1].unit.is_equivalent(mcube_in[0].units.y)):
+            
+            # convert the world to pixel
+            init_map = sunpy.map.Map(mcube_in[0])
+            c_x1, c_y1 = init_map.data_to_pixel(x1, y1)
+            c_x2, c_y2 = init_map.data_to_pixel(x2, y2)            
+                        
+        elif x1.unit.is_equivalent(u.pixel) and y1.unit.is_equivalent(u.pixel):
+            pass
```

Comment:
the variables `c_xx` are now only defined when the units are angular. You should de-quantify them here.